explorer/interpreter/interpreter.cpp

```diff
@@ -291,46 +293,89 @@ auto Interpreter::CreateStruct(const std::vector<FieldInitializer>& fields,
   return arena_->New<StructValue>(std::move(elements));
 }
 
-auto PatternMatch(Nonnull<const Value*> p, Nonnull<const Value*> v,
+static auto InitializePlaceholderValue(
+    const ValueNodeView& value_node, ExpressionResult v,
+    std::optional<Nonnull<RuntimeScope*>> bindings) {
+  switch (value_node.expression_category()) {
+    case ExpressionCategory::Reference:
+      if (v.expression_category() == ExpressionCategory::Value ||
+          v.expression_category() == ExpressionCategory::Reference) {
+        // Build by copying from value or reference expression
+        (*bindings)->Initialize(value_node, v.value());
+      } else /* ExpressionCategory::Initializing */ {
```

Comment:
```suggestion
      } else {
```

Since Carbon would not support this style of commenting, we are going without it, [per style](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting). If you want the case to be explicit, use a `switch`, since that would also validate that the enum is fully handled.

Similar for 321.